java/ql/src/Security/CWE/CWE-652/XQueryInjection.qhelp

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>The software uses external input to dynamically construct an XQuery expression used to retrieve data from an XML database, but it does not neutralize or incorrectly neutralizes that input. 
+This allows an attacker to control the structure of the query.</p>
```

Comment:
I think we can make this clearer by reorganizing it a little:

```suggestion
<p>The software uses external input to dynamically construct an XQuery expression which is then used to retrieve data from an XML database. 
However, the input is not neutralized, or is incorrectly neutralized, which allows an attacker to control the structure of the query.</p>
```

Reply:
@smowton What's your opinion?

Reply:
Agree with @felicitymay